pkg/reconciler/controlloop/pod.go

```diff
@@ -116,9 +116,6 @@ func (pc *PodController) Start(stopChan <-chan struct{}) {
 	}
 
 	go wait.Until(pc.worker, syncPeriod, stopChan)
-
-	<-stopChan
-	logging.Verbosef("shutting down network controller")
```

Comment:
This breaks the code-base; you need to add waiting on the stop channel to the main func.

Reply:
This does get addressed in the other commits - see latest code. I was trying to give this change its own commit - and in hindsight I agree adding a wait on stop channel in the main function makes this commit functional....but we don't expect or support someone to cherry pick this specific commit for their usage, so it should be fine this time.

Reply:
Yes, but why would you choose to add a broken commit ? 

This would trip tools like [`git bisect`](https://git-scm.com/docs/git-bisect), and I don't see any advantage to have it.
 
Just behavior is even explicitly listed as a no-go in projects like [ovn-k](https://github.com/ovn-org/ovn-kubernetes/blob/master/CONTRIBUTING.md#before-you-start). 

Do you think it is needed to do the same in whereabouts ?

Reply:
I think I may have originally misunderstood your comments about adding a separate commit for this. I did briefly consider the repercussions of a broken commit and figured it was inconsequential. I'll see if I can revise the commit to have the "wait for stopChan" line.